osu.Game/Screens/OsuScreen.cs

```diff
@@ -155,6 +190,12 @@ public override bool OnExiting(IScreen next)
             return false;
         }
 
+        private void updateActivity()
+        {
+            if (api != null)
+                api.LocalUser.Value.Activity.Value = activity;
```

Comment:
The activity is not set correctly if the user logs in on a screen that provided an activity.

I think a better flow is something like:

```
OsuScreen:
    updateActivity() => api.Activity.Value = activity;

APIAccess:
    LocalUser.ValueChanged => LocalUser.Activity.BindTo(activity);
```

Reply:
I'll update this.